Wikipedia/Code/WMFExploreSectionControllerCache.m

```diff
@@ -43,22 +50,80 @@ - (instancetype)initWithDataStore:(MWKDataStore*)dataStore {
         self.dataStore                              = dataStore;
         self.sectionControllersBySection            = [[NSCache alloc] init];
         self.sectionControllersBySection.countLimit = [WMFExploreSection totalMaxNumberOfSections];
-        self.sectionsBySectionController            = [NSMapTable mapTableWithKeyOptions:NSMapTableWeakMemory | NSMapTableObjectPointerPersonality
-                                                                            valueOptions:NSMapTableWeakMemory];
+
+        /*
+         Controllers (keys in the map) are retained by the NSCache and can be evicted at any time. On the other hand,
+         sections (values in the map) are "value types" which should be owned by the receiver.  In other words, sections
+         are created as needed, and not retained for any particular reason, and it would be unintentional for a value in
```

Comment:
@bgerstle can you explain why we would want this to retain the sections? This seems counter-intuitive. 

1. The cache is supposed to retain the controllers (and sections using its internal memory rules)
2. The map table only exists as a convenience - its only meant to replace the more verbose reverse look up code that previously existed.
3. I can't think of a reason that we would want the map table to retain a section if it has been released everywhere else.